src/amg/azext_amg/_params.py

```diff
@@ -0,0 +1,57 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=line-too-long
+
+
+def load_arguments(self, _):
+
+    from knack.arguments import CLIArgumentType
+    from azure.cli.core.commands.parameters import tags_type, get_three_state_flag
+    from azure.cli.core.commands.validators import get_default_location_from_resource_group
+    from ._validators import process_missing_resource_group_parameter
+
+    grafana_name_type = CLIArgumentType(options_list="--grafana-name",
+                                        help="Name of the Azure Managed Dashboard for Grafana.",
+                                        id_part="name")
+
+    with self.argument_context("grafana") as c:
+        c.argument("tags", tags_type)
+        c.argument("location", validator=get_default_location_from_resource_group)
+        c.argument("grafana_name", grafana_name_type, options_list=["--name", "-n"], id_part=None, validator=process_missing_resource_group_parameter)
+        c.argument("id", help=("The identifier (id) of a dashboard/data source is an auto-incrementing "
+                               "numeric value and is only unique per Grafana install."))
+        c.argument("folder", help="id, uid, title which can identify a folder. CLI will search in the order of id, uid, and title, till finds a match")
+
+    with self.argument_context("grafana create") as c:
+        c.argument("grafana_name", grafana_name_type, options_list=["--name", "-n"], validator=None)
+        c.argument("skip_system_assigned_identity", options_list=["-s", "--skip-system-assigned-identity"], arg_type=get_three_state_flag(), help="Do not enable system assigned identity")
+        c.argument("skip_role_assignments", arg_type=get_three_state_flag(), help="Do not create role assignments for managed identity and the current login user")
+
+    with self.argument_context("grafana delete") as c:
+        c.argument('yes', options_list=['--yes', '-y'], help='Do not prompt for confirmation.', action='store_true')
```

Comment:
When register commands with `confirmation=True` this argument will not need to state.

Reply:
updated